### PR TITLE
[4.0] Improving row selection in Joomla! Admin

### PR DESCRIPTION
**Pull request by jreys**
### Summary of Changes
Usually when I was using Joomla Admin, the multi-select on the list view was not very user-friendly since clicking on such a small square can sometimes fail on the first try. So I came up with a solution for 

#### Testing Instructions
- Go to articles (or any multi-select list view), try clicking anywhere on a row, it will select the whole row
- Try trashing multiple items by clicking severall rows
- Try the toggle-all button

### Expected result

When you click on a row on the articles list view (for example), it will highlight the whole line:
image

When you click on the select-all checkbox, it will highlight all items on that page:
image

### Notice

I'm aware of backwards compatibility so the regular checkbox will still work as before while still highlighting the whole line.

### CODE REVIEW

**Comment by dgt41** [comment]
Can you drop jquery (you don't need it for such easy task)  and use a class instead of changing directly the styles

**Comment by jreys** [comment]
> and use a class instead of changing directly the styles

Do you mean adding a class to the row when the action is done?

**Comment by C-Lodder** [comment]
- Chain variables
- No jQuery please
- Define your colour in the Atum `variables.scss`
- Add a class, apply background, call the colour
- Compile SCSS
- Use JS (`el.classList.toggle('foo');`) to toggle the class

**Comment by jreys** [comment]
- [ ] Chain variables
- [x] No jQuery please
- [x] Define your colour in the Atum variables.scss
- [x] Add a class, apply background, call the colour
- [ ] Compile SCSS
- [ ] Use JS (element.classList.toggle('foo');) to toggle the class

I have questions on the topics not checked: 
- What do you understand as Chain variables?
- how do I compile SCSS correctly?
- the last one seems obvious after SCSS is compiled tbh

**Comment by dgt41** [comment]
```js
"use strict";
var color = '#d9edf7', rows   = document.querySelectorAll('tr[class^="row"]');
```
The above snippet should take care of the first checkbox.

For compiling SCSS you need to have installed node and npm
Then in the root directory of joomla run npm install
Once complete run grunt compile

**Comment by C-Lodder** [comment]
Here are the instructions: https://github.com/joomla/joomla-cms/blob/4.0-dev/grunt-readme.md

File: administrator/templates/atum/scss/_variables.scss [hunk 1]
```diff
@@ -54,6 +54,7 @@ $fa-font-path:                     "../../../../media/vendor/font-awesome/fonts"
 // Tables
 $table-bg:                         transparent !default;
 $table-bg-accent:                  rgba(0,0,0,.03) !default;
+$table-row-selected                #d9edf7;
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/_variables.scss]: add a colon after the variable name, like so: `$table-row-selected:`

File: media/system/js/multiselect.js [hunk 2]
```diff
@@ -46,3 +46,48 @@ Joomla.JMultiSelect = function(table) {
 	};
 	initialize(table);
 };
+
+document.addEventListener("DOMContentLoaded", function(event) { 
+	color = '#d9edf7';
+	rows   = document.querySelectorAll('tr[class^="row"]');
```
Comment (C-Lodder) [inline media/system/js/multiselect.js]: change to:
```
var color = '#d9edf7',
      rows   = document.querySelectorAll('tr[class^="row"]');
```

File: media/system/js/multiselect.js [hunk 3]
```diff
@@ -46,3 +46,48 @@ Joomla.JMultiSelect = function(table) {
...
+		if (this.checked) {
+			rows.forEach(function(row, index) {
+				changeBg(row, color);
+			});
+		}
+		else {
+			rows.forEach(function(row, index) {
+				changeBg(row, '');
+			});
+		}
```
Comment (C-Lodder) [inline media/system/js/multiselect.js]: you can simplify this `if` statement to:

```
var color = this.checked ? color : '';

rows.forEach(function(row, index) {
	changeBg(row, colour);
});
```

**Comment by jreys** [comment]
I think that is all :)

**Comment by franz-wohlkoenig** [comment]
@jreys is this PR ready to Test?

**Comment by C-Lodder** [comment]
@jreys - one more thing, please use single quote's as opposed to double quotes where possible.

**Comment by jreys** [comment]
@franz-wohlkoenig it's ready for testing now, everything fixed

btw @C-Lodder I didn't use the `element.classList.toggle('foo');` because it was causing weird behaviours when you selected one row and then selected them all after, it would live the previous row unselected and it's not the wanted behaviour

**Review by C-Lodder — APPROVED** [review]

**Comment by C-Lodder** [comment]
@jreys - No worries. All seems to be good now.

@dgt41 - JS look ok?

**Comment by C-Lodder** [comment]
I have tested this item :white_check_mark: successfully on 0a5a3873c8dbd89c89ca2b2184d348cfbc21e7d4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14913">issues.joomla.org/tracker/joomla-cms/14913</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 0a5a3873c8dbd89c89ca2b2184d348cfbc21e7d4<br /><br />an optical Feedback like another Curser would help to see that i can move Column.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14913">issues.joomla.org/tracker/joomla-cms/14913</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14913">issues.joomla.org/tracker/joomla-cms/14913</a>.</sub>

**Comment by cokencorn** [comment]
I have tested this item :white_check_mark: successfully on ac8c87d224d0f1a6201ae5754dd20a767591216f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14913">issues.joomla.org/tracker/joomla-cms/14913</a>.</sub>

**Pull request merged**